src/blob_format.h

```diff
@@ -327,28 +330,34 @@ struct BlobFileHeader {
   static const uint32_t kHeaderMagicNumber = 0x2be0a614ul;
   static const uint32_t kVersion1 = 1;
   static const uint32_t kVersion2 = 2;
+  static const uint32_t kVersion3 = 3;
 
-  static const uint64_t kMinEncodedLength = 4 + 4;
-  static const uint64_t kMaxEncodedLength = 4 + 4 + 4;
+  static const uint64_t kV1EncodedLength = 4 + 4;
+  static const uint64_t kV2EncodedLength = 4 + 4 + 4;
+  static const uint64_t kV3EncodedLength = 4 + 4 + 4 + 8;
 
   // Flags:
   static const uint32_t kHasUncompressionDictionary = 1 << 0;
 
-  uint32_t version = kVersion2;
+  uint32_t version = kVersion3;
   uint32_t flags = 0;
+  uint64_t block_size = 0;
 
   static Status ValidateVersion(uint32_t ver) {
-    if (ver != BlobFileHeader::kVersion1 && ver != BlobFileHeader::kVersion2) {
+    if (ver != BlobFileHeader::kVersion1 && ver != BlobFileHeader::kVersion2 &&
+        ver != BlobFileHeader::kVersion3) {
       return Status::InvalidArgument("unrecognized blob file version " +
                                      ToString(ver));
     }
     return Status::OK();
   }
 
   uint64_t size() const {
-    return version == BlobFileHeader::kVersion1
-               ? BlobFileHeader::kMinEncodedLength
-               : BlobFileHeader::kMaxEncodedLength;
+    return version == BlobFileHeader::kVersion3
+               ? BlobFileHeader::kV3EncodedLength
+               : version == BlobFileHeader::kVersion2
+               ? ? BlobFileHeader::kV2EncodedLength
```

Comment:
duplicated '?'. Please add ( ) to explicitly show the operator priority.

Reply:
Done